onnxscript/function_libs/torch_lib/ops/nn.py

```diff
@@ -2013,7 +2005,55 @@ def _aten_scaled_dot_product_attention_no_mask_onnx(
     value: TFloat,
     scale: TFloat,
     dropout_p: float,
+    enable_gqa: bool,
 ) -> TFloat:
+    # Handle Grouped Query Attention (GQA) if enabled
+    if enable_gqa:
+        # Get head dimensions
+        query_shape = op.Shape(query)
+        key_shape = op.Shape(key)
+        query_heads = op.Slice(query_shape, [-3], [-2])  # query.size(-3)
+        key_heads = op.Slice(key_shape, [-3], [-2])      # key.size(-3)
+
+        # Calculate the repeat factor: query_heads // key_heads
+        repeat_factor = op.Div(query_heads, key_heads)
+
+        # Expand key and value to match query head dimension
+        # Implement key.repeat_interleave(repeat_factor, -3) using Expand
+        # First, get the shape of key and modify the head dimension
+        key_shape_expanded = op.Concat(
+            op.Slice(key_shape, [0], [-3]),  # batch and other dims
+            op.Mul(key_heads, repeat_factor),  # expanded head dimension
+            op.Slice(key_shape, [-2], [_INT64_MAX]),  # remaining dims
+            axis=0
+        )
+        # Expand key by repeating each head 'repeat_factor' times
+        key_unsqueezed = op.Unsqueeze(key, [-2])  # Add dimension for repeating
+        key_tiled = op.Tile(key_unsqueezed, op.Concat(
+            op.Constant(value_ints=[1, 1, 1]),  # don't repeat batch, seq, head dims
+            repeat_factor,  # repeat factor for the new dimension
+            op.Constant(value_ints=[1, 1]),  # don't repeat the remaining dims
+            axis=0
+        ))
+        key = op.Reshape(key_tiled, key_shape_expanded)
+
+        # Same for value
+        value_shape = op.Shape(value)
+        value_shape_expanded = op.Concat(
+            op.Slice(value_shape, [0], [-3]),  # batch and other dims
+            op.Mul(key_heads, repeat_factor),  # expanded head dimension
+            op.Slice(value_shape, [-2], [_INT64_MAX]),  # remaining dims
+            axis=0
+        )
+        value_unsqueezed = op.Unsqueeze(value, [-2])
+        value_tiled = op.Tile(value_unsqueezed, op.Concat(
```

Comment:
op.Tile does not align to PyTorch inplementation.

```python
    if (
        (q_num_heads != k_num_heads)
        and (q_num_heads % k_num_heads == 0)
        and (k_num_heads == v_num_heads)
    ):
        seq_reps = q_num_heads // k_num_heads
        # Interleave-repeat each KV head: [h0, h0, h1, h1, ...]
        K = np.repeat(K, repeats=seq_reps, axis=1)
        V = np.repeat(V, repeats=seq_reps, axis=1)
```

We should be able to reuse `repeat_interleave` here when it's done.